bs-platform@7.0.1 update

idkjs opened the pull request:
I have updated the package to `bs-platform@7.0.1`. There maybe a few places where we could remove the `.` notations where `js` objects are inlined, example in `AR.re` https://github.com/idkjs/reason-expo/blob/0e7a1f87eb6adbcbeb216b52667b8105179558dd/packages/reason-expo/src/AR.re#L522-L566

Not sure if the tests need to be adjusted. As of now, they all pass as is.

I have add new usage comments at the top of `AuthSession.re`. The old ones will have to be removed once accepted. https://github.com/idkjs/reason-expo/blob/e492bd72c98b7c4ca3bbc15a99d8da716d531359/packages/reason-expo/src/AuthSession.re#L14-L22

Looking forward you your feedback to see how to proceed.

Peace/Love.

peterpme commented:
Hey @idkjs so sorry for the delay! Is this still good to go?

idkjs commented:
Hi @peterpme. I would have to go back over it since we are up to `bs-platform@7.2.2` and there were some breaking changes for some of the syntax that we came across in other repos. I will go ahead and do that. Yes?

peterpme commented:
That would be great, thank you so much @idkjs !

idkjs commented:
@peterpme this last commit updates the deps to the following: 
```json
"dependencies": {
    "expo": "~36.0.0",
    "react": "~16.9.0",
    "react-dom": "~16.9.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz",
    "reason-expo": "^36.0.0",
    "react-native-web": "~0.11.7",
    "reason-react-native": "^0.61.1",
    "reason-react": "^0.7.0"
  },
  "devDependencies": {
    "@babel/core": "^7.0.0",
    "babel-preset-expo": "~8.0.0",
    "bs-platform": "^7.2.2",
    "expo-yarn-workspaces": "^1.2.1"
  },
```
In `packages/test` `rn-cli.config.js` becomes `metro.config.js` per expo `expo-yarn-workspaces`.

I moved around the types in `ImageManipulator.re` and `Camera.re` to get rid of the bucklescript warnings.

`.gitAttributes` updates `OCaml` to `Reason`.

I feel like there are too many deps in https://github.com/idkjs/reason-expo/blob/cf55209ba848248702446c82fabd4b2bc24bcec6/packages/reason-expo/package.json#L24-L41 based on what was there before but that is what I was left with after working through the build errors. Maybe you have some insight there.

Using `sdk-36.0.0`.

The demo runs on web:
<img width="1159" alt="Screen Shot 2020-03-24 at 5 45 06 PM" src="https://user-images.githubusercontent.com/2370391/77480129-84c3de00-6df7-11ea-80d8-7db832f8583a.png">

And Ios:
![Simulator Screen Shot - iPhone 11 Pro Max - 2020-03-24 at 17 46 10](https://user-images.githubusercontent.com/2370391/77480160-94432700-6df7-11ea-9c94-97f338d746da.png)


Not sure if that is what you all wanted to do. Thoughts?

peterpme commented:
so far looks great! just a couple merge conflicts

idkjs commented:
So work with me, brother. What is the protocol for resolving merge conflicts? Is that for me to do? if so how do I see the conflict? I'm greyed out in this ui.

peterpme commented on packages/template/package.json:
```diff
@@ -11,16 +11,18 @@
-    "react-native": "https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz",
-    "reason-expo": "^34.0.0",
+    "expo": "~36.0.0",
+    "react": "~16.9.0",
```
```suggestion
    "react": "16.9.0",
```

needs to be hardcoded

peterpme commented on packages/template/package.json:
```diff
@@ -11,16 +11,18 @@
-    "react": "16.8.3",
-    "react-native": "https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz",
-    "reason-expo": "^34.0.0",
+    "expo": "~36.0.0",
```
```suggestion
    "expo": "36.0.0",
```

needs to be hardcoded

peterpme commented on packages/test/.expo-shared/assets.json:
```diff
@@ -0,0 +1,4 @@
+{
```
whats this?

idkjs replied:
This is a generated file. I did not touch it.

peterpme commented:
Hey @idkjs, I actually just noticed that as well! After some investigation, I'm not sure why `.gitattributes` is a conflict because it doesn't exist.

Is it easy for you to pull down master and then merge it into your branch?

For both `.gitattributes` and `packages/reason-expo/package.json` override your new version. I looked into how Expo's new stuff works and it aligns with what you're doing.

That should make fixing the merge conflict easy!

idkjs commented:
It would be easy if I new how to do it. Seems like if I do that, it will override all everything I updated, no? I could just change the `.gitattributes` manually but not sure what the problem is with `package.json`. Any guidance? Thanks.

peterpme commented:
I don't think so:

on your local fork:
```
git checkout master
git pull
git checkout idkjs:bs_platform_7
git merge master
```

it should be as that? maybe i'm missing something?

idkjs commented:
Ran your commands:

```sh
~/G/reason-expo ❯❯❯ git branch
* bs_platform_7
  master
~/G/reason-expo ❯❯❯ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
~/G/reason-expo ❯❯❯ git pull
Already up to date.
~/G/reason-expo ❯❯❯ git checkout bs_platform_7
Switched to branch 'bs_platform_7'
Your branch is up to date with 'origin/bs_platform_7'.
~/G/reason-expo ❯❯❯ git merge master
Auto-merging packages/reason-expo/src/SQLite.re
Auto-merging packages/reason-expo/package.json
CONFLICT (content): Merge conflict in packages/reason-expo/package.json
CONFLICT (modify/delete): .gitattributes deleted in master and modified in HEAD. Version HEAD of .gitattributes left in tree.
Automatic merge failed; fix conflicts and then commit the result.
~/G/reason-expo ❯❯❯                                                                                                         ✘ 1 
```

So how do we fix these conflicts?

peterpme commented on packages/reason-expo/package.json:
```diff
@@ -1,7 +1,11 @@
 {
   "name": "reason-expo",
   "description": "ReasonML bindings for Expo",
+<<<<<<< HEAD
```
can you remove the <<<< and >>> and only keep the "version" key at 36?

peterpme commented on .gitattributes:
```diff
@@ -1,2 +0,0 @@
-*.re linguist-language=Reason
-*.rei linguist-language=Reason
```
i think you want to re-add this file

peterpme commented on packages/reason-expo/src/SQLite.re:
```diff
@@ -82,5 +82,5 @@ module Database = {
 };
 
-[@bs.module "expo-sqlite"] [@bs.scope "SQLite"]
+[@bs.module "expo-sqlite"]
```
are you supposed to remove bs.scope? if so this looks good

idkjs replied:
removed [@bs.scope "SQLite"] because failed on testing. Works like this.

peterpme commented on packages/reason-expo/src/Svg.re:
```diff
@@ -167,6 +167,7 @@ module G = {
       ~fill: string=?,
       ~stroke: string=?,
       ~strokeWidth: string=?,
+      ~transform: string=?,
```
are you adding or removing transform?

idkjs replied:
Looks like it should be there and it in both versions. Maybe it had been removed erroneously and re-added.

peterpme commented:
Hey @idkjs nice! Really close. One merge conflict in package.json. Then a few questions pertaining to the Svg.re file and .gitattributes that was deleted

idkjs commented:
I think we are good now. Other issues?

Thank you.

zth commented:
Nice work @idkjs , looking forward to using this once it lands!

sagarjs commented:
any updates on this pull request?
looking forward to the merge

peterpme commented:
Hi sorry about this delay, thank you @idkjs !!!
